src/Informer/CMakeLists.txt

```diff
@@ -34,6 +34,13 @@ spectre_target_headers(
   Verbosity.hpp
   )
 
+# The Darwin linker is not OK with undefined symbols at link time so we need to
+# appease it. This is needed so compiling InfoAtLink.cpp can be deferred to link
+# time.
```

Comment:
Could you expand on the problem?  In particular, this currently reads as "The symbols can't be undefined at link time, but we want to define them at link time," which doesn't sound like a problem.  (As a code comment if this doesn't change significantly, but I'm trying to understand what is going on.)  I feel like I've seen something like this in the past, but can't remember any details.

Reply:
Reworded the comment. We want to wait with defining the function until we link the "end product", i.e. the executable.

@nilsdeppe Is the reason for this _only_ that we determine the time the _executable_ is linked, as opposed to the time the _library_ is linked? Is that really worth having these compiler wrappers, the extra logic in the build system, potential issues with compilers complaining about undefined symbols (this PR) and dummy-implementations of `info_at_build` (e.g. in Pybindings)?

Reply:
You've just pointed out the issue in your question :) You are assuming O(minutes), but it can be days, or even weeks since if no dependency of the lib is updated, the lib won't ever be relinking, while the executable must be. We also can't have every executable define the symbol at link time because that's either multiple definitions or UB (multiple weak references with differing implementations, no diagnostic required). If you have a way of getting this info into the executables and making sure it's up-to-date that's simpler I'm happy to change to it, but this was the only safe way we could figure out at the time. Oh, and having CMake set these things doesn't work because they you have to re-run CMake every time anything happens (I really mean almost anything), which is really annoying and what was done before the current implementation. It wasted _a lot_ of time

Reply:
Ah of course, thanks for explaining!